src/_pytest/unittest.py

```diff
@@ -153,6 +154,11 @@ def _addexcinfo(self, rawexcinfo):
         self.__dict__.setdefault("_excinfo", []).append(excinfo)
 
     def addError(self, testcase, rawexcinfo):
+        try:
+            if isinstance(rawexcinfo[1], exit.Exception):
+                exit(rawexcinfo[1].msg)
+        except TypeError:
+            pass
```

Comment:
This is ugly, but some test passes in 42 here, and I did not feel like having it in `self._addexcinfo`.